### PR TITLE
SQL-974: Support running local ADF instance on Windows

### PR DESCRIPTION
**Pull request by bucaojit**
Tested on `windows-vsCurrent-small` and `windows-64-vs2019-small` with the cygwin shell.

### CODE REVIEW

File: resources/run_adl.sh [hunk 1]
```diff
@@ -146,25 +163,35 @@ check_mongod
...
     mkdir -p $TMP_DIR
 
-    # Note: ADL has a storage.json file that generates configs for us.
-    # The mongodb source is on port $MONGOD_PORT so we use that here.
-    $MONGO_DOWNLOAD_DIR/bin/mongod --port $MONGOD_PORT --dbpath $MONGO_DB_PATH \
-      --logpath $LOGS_PATH/mongodb_test.log --pidfilepath $TMP_DIR/${MONGOD}.pid --fork
+    # Install and start mongod
+    (cd $LOCAL_INSTALL_DIR && curl -O $MONGO_DOWNLOAD_LINK)
+
+    # Uncompressed the archive
```
Comment (mattChiaravalloti) [inline resources/run_adl.sh:174]:
```suggestion
    # Uncompress the archive
```
typo

File: resources/run_adl.sh [hunk 2]
```diff
@@ -56,6 +63,8 @@ MONGO_DOWNLOAD_UBUNTU=mongodb-linux-x86_64-ubuntu1804-5.0.4.tgz
 MONGO_DOWNLOAD_REDHAT=mongodb-linux-x86_64-rhel70-5.0.4.tgz
 # macOS
 MONGO_DOWNLOAD_MAC=mongodb-macos-x86_64-5.0.4.tgz
+# Windows
+MONGO_DOWNLOAD_WIN=mongodb-windows-x86_64-5.0.4.zip
```
Comment (mattChiaravalloti) [inline resources/run_adl.sh:67]: I wonder if/when we want to bump these versions from 5.0 to something later (a rapid release or 6.0). Just a thought; no action needed on this PR.

**Review by mattChiaravalloti — APPROVED** [review]
looks good! 🪟

File: resources/run_adl.sh (hunk 2 above)
Comment (bucaojit) [reply resources/run_adl.sh:67]: I was thinking the same when I added this line, whether to update these to 5.0.10.   One thought I had was to use curator for the downloads which would at least get the updated rapid release.  
I can create an investigation ticket to track this and find how we want to handle this going forward.

File: resources/run_adl.sh [hunk 3]
```diff
@@ -146,25 +163,35 @@ check_mongod
...
+	          | cut -d ' ' -f 5 | cut -d/ -f1)
+      chmod +x $LOCAL_INSTALL_DIR/$MONGO_UNZIP_DIR/bin/mongod.exe
+      MONGO_DOWNLOAD_DIR=$LOCAL_INSTALL_DIR/$MONGO_UNZIP_DIR
+      # mongod does not have --fork option on Windows, using nohup
+      nohup $MONGO_DOWNLOAD_DIR/bin/mongod --port $MONGOD_PORT --dbpath $(cygpath -m ${MONGO_DB_PATH}) \
+              --logpath $(cygpath -m $LOGS_PATH/mongodb_test.log) &
+      echo $! > $TMP_DIR/${MONGOD}.pid
+    else
+      tar zxvf $LOCAL_INSTALL_DIR/$MONGO_DOWNLOAD_FILE --directory $LOCAL_INSTALL_DIR
+      # Note: ADL has a storage.json file that generates configs for us.
```
Comment (nbagnard) [inline resources/run_adl.sh:189]: Do you want to move this comment up, above the if? It seems to apply to both section.

**Review by nbagnard — APPROVED** [review]
Just 2 very small suggestions.

File: resources/run_adl.sh [hunk 4]
```diff
@@ -29,6 +29,12 @@ if [ -d "/opt/golang/$GO_VERSION" ]; then
   GOROOT="/opt/golang/$GO_VERSION"
   GOBINDIR="$GOROOT"/bin
   PATH=$GOBINDIR:$PATH
+elif [ -d "C:\golang\\$GO_VERSION" ]; then
+  GOROOT="C:\golang\\$GO_VERSION"
+  GOBINDIR="$GOROOT"\\bin
+  PATH=$GOBINDIR:$PATH
```
[the remark this answers is not in the document]
Comment (bucaojit) [reply resources/run_adl.sh]: Made this change

File: resources/run_adl.sh (hunk 3 above)
Comment (bucaojit) [reply resources/run_adl.sh]: Sounds good, I moved it up.

**Comment by bucaojit** [comment]
evergreen merge

**Pull request merged**